### ISSUE TITLE
Feature Request: Allow Option to Not Autocomplete on Enter

### ISSUE THREAD

**Issue by Borthralla** [notepad-plus-plus/notepad-plus-plus#4799]
While I like autocomplete, it sometimes gets in the way. When I'm at the end of the line, I press enter as soon as I'm finished typing the last word. However, very often this inadvertently triggers the autocomplete which means I have to stop typing, drag my mouse up to the previous line, highlight the word, retype the original word, and then press escape and then enter, This takes way too long and is a serious interruption to my train of thought. 

I tried to fix this by going into the scintilla shortcuts and changing sci_newline from enter to shift+enter. Although this fixed my autocomplete issue, it also had the unintended consequence of notepad++ interpreting enter as only carriage-return. Pressing enter no longer put a newline! That caused auto-indentation to fail and my code to be broken.

What's more, when I tried to remap sci_enter back to enter, I was unable to because enter was not a valid option. At that point I just uninstalled and reinstalled notepad++.

My suggestion: Add an option to not auto-complete on enter (just tab). Also, it probably would be a good idea to allow sci_enter to be bound to enter.

**Comment by endolith** [comment]
Yes this is super annoying when trying to create a newline.  I use Ctrl-Z then Enter.  For every newline that ends in a letter.  How do people live with this??

No solution here either: https://superuser.com/q/1048224/13889

**Comment by calebjely** [comment]
Agreed, I have the same issue often. Autocomplete should only be triggered by TAB not ENTER.

**Comment by sasumner** [comment]
>How do people live with this??

We turn autocomplete OFF.  Seriously, autocomplete is not usable in Notepad++.

**Comment by 5j9** [comment]
https://superuser.com/a/1057259/340929 suggests that there might be a simple solution for this issue:

> It can be fixed by removing the following three lines from src/ScintillaBase.cxx:

>         case SCI_NEWLINE:
>             AutoCompleteCompleted(0, SC_AC_NEWLINE);
>             return 0;

**Comment by blaisemGH** [comment]
@Uhf7 proposed a sensible long-term solution for the standard distribution [in this comment](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/8389#issuecomment-640685638).
